packages/poller/src/invoice/processInvoices.ts

```diff
@@ -14,75 +13,397 @@ import {
   getCustodiedBalances,
 } from '../helpers';
 import { isValidInvoice } from './validation';
-import { ChainService } from '@mark/chainservice';
-import { Web3Signer } from '@mark/web3signer';
-import { Wallet } from 'ethers';
-
-interface ProcessInvoicesParams {
-  invoices: Invoice[];
-  cache: PurchaseCache;
-  logger: Logger;
-  everclear: EverclearAdapter;
-  chainService: ChainService;
-  prometheus: PrometheusAdapter;
-  config: MarkConfiguration;
-  web3Signer: Web3Signer | Wallet;
-}
+import { PurchaseAction } from '@mark/cache';
 
 export const MAX_DESTINATIONS = 10; // enforced onchain at 10
 export const TOP_N_DESTINATIONS = 7; // mark's preferred top-N domains ordered in his config
 
 const getTimeSeconds = () => Math.floor(Date.now() / 1000);
 
-export async function processInvoices({
-  invoices,
-  everclear,
-  cache,
-  chainService,
-  logger,
-  prometheus,
-  config,
-  web3Signer,
-}: ProcessInvoicesParams): Promise<void> {
-  // Create request id tracker for polling
-  const requestId = hexlify(randomBytes(32));
-
-  // Get current time to measure invoice age
-  const time = getTimeSeconds();
-  let start = time;
-  logger.debug('Method start', { requestId, start });
+export interface TickerGroup {
+  ticker: string;
+  invoices: Invoice[];
+  remainingBalances: Map<string, Map<string, bigint>>;
+  remainingCustodied: Map<string, Map<string, bigint>>;
+  chosenOrigin: string | null;
+}
+
+interface ProcessTickerGroupResult {
+  purchases: PurchaseAction[];
+  remainingBalances: Map<string, Map<string, bigint>>;
+  remainingCustodied: Map<string, Map<string, bigint>>;
+}
+
+interface InvoiceWithIntents {
+  invoice: Invoice;
+  intents: NewIntentParams[];
+  totalAllocated: bigint;
+}
+
+interface BatchedTickerGroup {
+  ticker: string;
+  origin: string;
+  invoicesWithIntents: InvoiceWithIntents[];
+  totalIntents: number;
+}
 
+/**
+ * Groups invoices by ticker hash
+ * @param context - The processing context
+ * @param invoices - The invoices to group
+ * @returns A map of ticker hash to invoices, sorted by oldest first
+ */
+export function groupInvoicesByTicker(
+  context: ProcessingContext,
+  invoices: Invoice[]
+): Map<string, Invoice[]> {
+  const { prometheus } = context;
+
+  const invoiceQueues = new Map<string, Invoice[]>();
+  
+  invoices
+    .sort((a, b) => a.hub_invoice_enqueued_timestamp - b.hub_invoice_enqueued_timestamp)
```

Comment:
will we always want the sorting to be done this way (oldest to newest)

Reply:
Yeah based on algo below